### PR TITLE
Feature/permissions modal add computed perms

### LINKED ISSUE #1127: Permission modal: improve the permission display (add computed perms)
## Overview
Improve the permission display by adding the computed permissions.

Require #1125 to be completed.

## Acceptance criteria

For each permission shown in the permission modal:

- [ ] Don't use a "lock" icon but a tick icon for selected permission.
- [ ] If the "computed" permission (from the [view perm service](https://franceioi-algorea.s3.eu-west-3.amazonaws.com/spec/v1.2.0.html#tag/groups/operation/permissionsView)) is higher than the currently selected one, show all the "circles" from the one following the selected one to the "computed" one in light blue with the tick icon.
For instance: 
Screenshot 2022-08-04 at 20 24 34
--> if the computed perm on can_view is "content and descendants", the small circle on the left of "content" and "content and contestants" have to be displayed in light blue with a tick inside (same as "nothing" but in a lighter color). The line between the "circles" can stay gray.

- [ ] When hovering these lighter blue circles, the tooltip should show (in addition, on a new line if there was already a message) "This group has this permission via one of one of its ancestor group."/"This user has this permission via one of the other group he is member of, or one of their ancestors.."

### PR DESCRIPTION
**Pull request by Iloveall**
## Description

Fixes #1127

## Notes (out of scope, known isues, hints for reviewing code, ...)  (optional)

...

## Test cases

- [ ] Case 1:
  1. Given I am the usual user
  2. When I go to [this page](https://dev.algorea.org/branch/feature/permissions-modal-add-computed-perms/en/groups/by-id/6710944276987033666;path=5121055722873780306/details)
  3. And I click on start watch
  4. Then I click on "Item Title (Pixal)" of suggested list
  5. And I click on permissions section and "Edit permissions" button
  6. Then I click on "Can view" section
  7. Then I see Info, Content, Content and descendants as ticks and with light blue color
  8. Then I hover on one of the tick
  9. Then I see tooltip with message: "This group has this permission via one of one of its ancestor group." 

- [ ] Case 2:
  1. Given I am the usual user
  2. When I go to [this page](https://dev.algorea.org/branch/feature/permissions-modal-add-computed-perms/en/groups/by-id/6710944276987033666;path=5121055722873780306/details)
  3. And I click on start watch
  4. Then I click on "Item Title (Pixal)" of suggested list
  5. And I click on "Progress" -> "Chapter view" tab
  6. Then I click on one of score rings of "usr_5p020x2thuyu" user and I click on "Access" button in popover
  7. Then I click on "Can view" section
  8. Then I see Info, Content, Content and descendants as ticks and with light blue color
  9. Then I hover on one of the tick
  10. Then I see tooltip with message: "This user has this permission via one of the other group he is member of, or one of their ancestors." 

- [ ] Case 3:
  1. Given I am the usual user
  2. When I go to [this page](https://dev.algorea.org/branch/feature/permissions-modal-add-computed-perms/en/groups/by-id/6710944276987033666;path=5121055722873780306/details)
  3. And I click on start watch
  4. Then I click on "Item Title (Pixal)" of suggested list
  5. And I click on "Progress" -> "Chapter view" tab
  6. Then I click on one of score rings of "usr_5p020x2thuyu" user and I click on "Access" button in popover
  7. Then I click on "Can grant view" section
  8. Then I see Info and enter, Content, Content and descendants, Solution, Solution and grant as ticks and with light blue color
  9. Then I hover on one of the tick
  10. Then I see tooltip with messages: "This user has this permission via one of the other group he is member of, or one of their ancestors." , "This user needs...."

- [ ] Case 3:
  1. Given I am the usual user
  2. When I go to [this page](https://dev.algorea.org/branch/feature/permissions-modal-add-computed-perms/en/groups/by-id/6710944276987033666;path=5121055722873780306/details/managers)
  3. And I click on edit permissions of "Armelle Bonenfant (arbonenfant)"
  4. Then I see "Read-only", "Membership" with checked ticks

### CODE REVIEW

**Comment by smadbe** [comment]
The first item of the spec is not done "Don't use a "lock" icon but a tick icon for selected permission." 
Screenshot 2022-10-20 at 10 09 11

File: src/app/modules/item/helpers/computed-permissions.ts [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+
+export function withComputePermissions(
+  permissionsDialogData: PermissionsDialogData,
+  permissions: Omit<GroupPermissions,'canEnterFrom'|'canEnterUntil'>,
+  receiverPermissions: Omit<GroupPermissions,'canEnterFrom'|'canEnterUntil'>,
+  computedPermissions: Omit<GroupComputedPermissions,'canEnterFrom'|'canEnterUntil'>,
+  targetType: TypeFilter,
+): PermissionsDialogData {
+  const giverCanAtLeastView = hasHigherPermission(itemViewPermValues, computedPermissions.canView);
+  const giverCanAtLeastGrandView = hasHigherPermission(itemGrantViewPermValues, computedPermissions.canGrantView);
```
Comment (smadbe) [inline src/app/modules/item/helpers/computed-permissions.ts:53]: grand -> grant

**Review by smadbe — APPROVED** [review]
👍  Looks good. Only a few details on the code.

File: src/app/modules/item/helpers/computed-permissions.ts (hunk 1 above)
Comment (Iloveall) [reply src/app/modules/item/helpers/computed-permissions.ts:53]: Fixed

File: src/app/modules/shared-components/components/collapsible-section/progress-select/progress-select.component.html [hunk 2]
```diff
@@ -14,8 +14,16 @@
...
+      <span
+        class="progress-point"
+        [ngClass]="{active: idx < selected + 1, last: idx + 1 == selected + 1, 'active-light': item.selected}"
+        (click)="onSet(item.value)"
+        [pTooltip]="item.tooltip?.join('\n') || ''"
+        tooltipPosition="right"
+        tooltipStyleClass="permissions-dialog"
+        [escape]="false"
+      >
+        <i *ngIf="type === 'checksWithLock' && (idx < selected || item.selected)" class="fa fa-check"></i>
```
[the remark this answers is not in the document]
Comment (Iloveall) [reply src/app/modules/shared-components/components/collapsible-section/progress-select/progress-select.component.html]: Fixed

**Pull request merged**